pxtblocks/fields/field_gridpicker.ts

```diff
@@ -250,10 +260,16 @@ namespace pxtblockly {
                 }, 300, false));
 
                 searchBar.addEventListener("keyup", (e) => {
-                    let firstItem = this.getFirstItem.bind(this)()
-                    if (e.keyCode == 13 && firstItem) {
-                        this.selectItem.bind(this)(firstItem)
+                    if (e.keyCode == 13) {
+                        let text = searchBar.value;
+                        let firstItem = this.getFirstItem()
+                        if (text && firstItem) {
+                            this.selectItem(firstItem)
+                        } else {
+                            this.close()
```

Comment:
I think "enter" should not close the picker if there are no blocks to select, I think it's ok to just leave the picker open

Reply:
Done